File: JumpToEnd/JumpToEnd.py
def JumpToEnd(arr):
    newMax = 0
    jump = 0
    oldMax = 0
    for i in range(0 ,len(arr)-1):
        newMax = max(newMax , i+arr[i])
        if(i == oldMax):
            jump += 1
            oldMax = newMax
    print(jump)



JumpToEnd([1 , 2  , 3 , 5 , 8 , 9 , 1 ,])

File: JumpToEnd/JumpToEnd.js
function jumpToEnd(nums) {
  let newMax = 0;
  let jump = 0;
  let oldMax = 0;
  for (let i=0;i<nums.length-1;i++) {
      newMax = Math.max(newMax, i+nums[i]);
      if (i == oldMax) {
          jump++;
          oldMax = newMax;
      }
  }
  console.log(jump) ;
};


jumpToEnd([1 , 2  , 3 , 5 , 8 , 9 , 1 ,])